cloud/deployment/deployment.go

```diff
@@ -274,103 +288,274 @@ func Create(label, workspaceID, description, clusterID, runtimeVersion, dagDeplo
 	}
 
 	// select and validate cluster
-	clusterID, err = useSharedClusterOrSelectDedicatedCluster(cloudProvider, region, c.Organization, clusterID, clusterType, coreClient)
-	if err != nil {
-		return err
-	}
-
-	scheduler := astro.Scheduler{
-		AU:       schedulerAU,
-		Replicas: schedulerReplicas,
-	}
-
-	spec := astro.DeploymentCreateSpec{
-		Executor:  executor,
-		Scheduler: scheduler,
+	if !IsDeploymentStandard(deploymentType) {
+		clusterID, err = selectCluster(clusterID, c.Organization, platformCoreClient)
+		if err != nil {
+			return err
+		}
 	}
 
-	if dagDeploy == "enable" { //nolint: goconst
+	if dagDeploy == enable { //nolint: goconst
 		dagDeployEnabled = true
 	} else {
 		dagDeployEnabled = false
 	}
+	createDeploymentRequest := astroplatformcore.CreateDeploymentRequest{}
+	if IsDeploymentStandard(deploymentType) || IsDeploymentDedicated(deploymentType) {
+		var highAvailabilityValue bool
+		if highAvailability == enable { //nolint: goconst
+			highAvailabilityValue = true
+		} else {
+			highAvailabilityValue = false
+		}
```

Comment:
```suggestion
highAvailabilityValue := highAvailability == enable
```

Reply:
highAvailability is a string and highAvailabilityValue is a bool